pkg/securityscan/core/configmap.go

```diff
@@ -142,9 +143,29 @@ func getConfigMapObject(cmName, data string) *corev1.ConfigMap {
 	}
 }
 
-func getCustomBenchmarkConfigMap(benchmark *cisoperatorapiv1.ClusterScanBenchmark, configmapsClient wcorev1.ConfigMapController) (*corev1.ConfigMap, error) {
+func getCustomBenchmarkConfigMap(benchmark *cisoperatorapiv1.ClusterScanBenchmark, clusterscan *cisoperatorapiv1.ClusterScan, configmapsClient wcorev1.ConfigMapController) (*corev1.ConfigMap, error) {
 	if benchmark.Spec.CustomBenchmarkConfigMapName == "" {
 		return nil, nil
 	}
-	return configmapsClient.Get(benchmark.Spec.CustomBenchmarkConfigMapNamespace, benchmark.Spec.CustomBenchmarkConfigMapName, metav1.GetOptions{})
+	userConfigmap, err := configmapsClient.Get(benchmark.Spec.CustomBenchmarkConfigMapNamespace, benchmark.Spec.CustomBenchmarkConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	if benchmark.Spec.CustomBenchmarkConfigMapNamespace == cisoperatorapiv1.ClusterScanNS {
+		return userConfigmap, nil
+	}
+	//copy the configmap to ClusterScanNS so that cis scan pod can find it for volume mount
+	//this will be cleaned up after scan job finishes
```

Comment:
If the user uploads the custom benchmark configmap to cis-operator-system, scan pods can find it. But if some other namespace is used by the user, here we create a copy of the configmap in cis-operator-system namespace so that volume mount finds the configmap.

Reply:
I don't know the cis-roles in detail, but with this it looks like user who does not have access to the cis-operator-system namespace can also create a configmap in some other namespace and add/modify a custom benchmark?

Reply:
A custombenchmark can be created with a configmap only by a  admin/ cluster-owner or a user with cis-admin role . A regular user cannot create a custombenchmark.